platform/web/detect.py

```diff
@@ -77,26 +77,18 @@ def configure(env):
         sys.exit(255)
 
     ## Build type
-    if env["target"].startswith("release"):
-        if env["optimize"] == "size":
-            env.Append(CCFLAGS=["-Os"])
-            env.Append(LINKFLAGS=["-Os"])
-        elif env["optimize"] == "speed":
-            env.Append(CCFLAGS=["-O3"])
-            env.Append(LINKFLAGS=["-O3"])
-
-        if env["target"] == "release_debug":
-            # Retain function names for backtraces at the cost of file size.
-            env.Append(LINKFLAGS=["--profiling-funcs"])
-    else:  # "debug"
-        env.Append(CCFLAGS=["-O1", "-g"])
-        env.Append(LINKFLAGS=["-O1", "-g"])
```

Comment:
@Faless By unifying the code with other platforms a couple changes were made here:
- `optimize=speed` now uses `-O3` for `template-release` but `-O2` for `template-debug`/`editor`, like on other platforms. We optimize for size by default anyway so I figured this isn't a big deal?
- `-dev` builds (ex `target=debug`) default to `debug` optimization level i.e. `-Og`, instead of `-O1` as was set here. It seemed to me the change is fine as `debug` is not something we ship, and `-Og` is supposed to be fairly close to `-O1`, with the exception of those optimizations that hinder debugging.